runtime/ops/web_worker/sync_fetch.rs

```diff
@@ -0,0 +1,182 @@
+              .headers()
+              .get("Content-Type")
+              .and_then(|v| v.to_str().ok())
+              .map(mime_type_essence);
+
+            // Always check the MIME type with HTTP(S).
+            loose_mime_checks = false;
+
+            let body = resp.bytes().await?;
+
+            (body, mime_type, res_url)
+          }
+          "data" => {
+            let data_url = DataUrl::process(&script)
+              .map_err(|e| type_error(format!("{:?}", e)))?;
+
+            let mime_type = {
+              let mime = data_url.mime_type();
+              format!("{}/{}", mime.type_, mime.subtype)
+            };
+
+            let (body, _) = data_url
+              .decode_to_vec()
+              .map_err(|e| type_error(format!("{:?}", e)))?;
+
+            (Bytes::from(body), Some(mime_type), script)
+          }
+          "blob" => {
+            let blob = blob_store
+              .get_object_url(script_url)?
+              .ok_or_else(|| type_error("Blob for the given URL not found."))?;
+
+            let mime_type = mime_type_essence(&blob.media_type);
+
+            let body = blob.read_all().await?;
+
+            (Bytes::from(body), Some(mime_type), script)
+          }
+          _ => {
+            return Err(type_error(format!(
+              "Classic scripts with scheme {}: are not supported in workers.",
+              script_url.scheme()
+            )))
+          }
+        };
+
+        if !loose_mime_checks {
+          // TODO(andreubotella) Check properly for a Javascript MIME type.
+          match mime_type.as_deref() {
+            Some("application/javascript" | "text/javascript") => {}
+            Some(mime_type) => {
+              return Err(
+                DomExceptionNetworkError {
+                  msg: format!("Invalid MIME type {:?}.", mime_type),
+                }
+                .into(),
+              )
+            }
+            None => {
+              return Err(
+                DomExceptionNetworkError::new("Missing MIME type.").into(),
+              )
+            }
+          }
+        }
+
+        let (text, _) = encoding_rs::UTF_8.decode_with_bom_removal(&body);
+
+        Ok(SyncFetchScript {
+          url: res_url,
+          script: text.into_owned(),
+        })
+      })
+    })
+    .collect();
+
+  let mut ret = Vec::with_capacity(handles.len());
+  for handle in handles {
+    let script = runtime.block_on(handle)??;
+    ret.push(script);
+  }
+  Ok(ret)
+}
+
+#[derive(Debug)]
+pub struct DomExceptionNetworkError {
+  pub msg: String,
+}
+
+impl DomExceptionNetworkError {
+  pub fn new(msg: &str) -> Self {
+    DomExceptionNetworkError {
+      msg: msg.to_string(),
+    }
+  }
+}
+
+impl std::fmt::Display for DomExceptionNetworkError {
+  fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+    f.pad(&self.msg)
+  }
+}
+
+impl std::error::Error for DomExceptionNetworkError {}
+
+pub fn get_dom_exception_network_error_class_name(
```

Comment:
Can you move this into the ext/web crate? The other Rust wrappers for dom_exception are defined there.

Reply:
Since I opened this PR, Deno gained an implementation of the `NetworkError` `DomException`, since it's used in `WebSocketStream`, so I'll just remove those wrappers and use the existing implementation.

Reply:
Oh perfect!